Airline_Web_API/Airline_Web_API/Controllers/TicketTypesController.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Airline_Web_API.Helpers;
+using Airline_Web_API.Services;
+using Airline_Web_API.ViewModels;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore.Storage.ValueConversion.Internal;
+
+namespace Airline_Web_API.Controllers
+{
+    [Route("api/ticket-types")]
+    [ApiController]
+    public class TicketTypesController : ControllerBase
+    {
+        private readonly TicketTypeService _ticketTypeService;
+        public TicketTypesController(TicketTypeService ticketTypeService)
+        {
+            _ticketTypeService = ticketTypeService;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<TicketTypeViewModel>>> GetTicketTypes()
+        {
+            var ticketTypes = await _ticketTypeService.GetTicketTypesAsync();
+
+            if (ticketTypes == null)
+            {
+                return NoContent();
+            }
+
+            return Ok(ticketTypes);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<TicketTypeViewModel>> GetTicketType(int id)
+        {
+            var ticketType = await _ticketTypeService.GetTicketTypeByIdAsync(id);
+
+            if (ticketType == null)
+            {
+                return NoContent();
+            }
+
+            return Ok(ticketType);
+        }
+
+        [Authorize(Roles = "admin")]
+        [HttpPost]
+        public async Task<ActionResult> Post([FromBody] TicketTypeViewModel model)
+        {
+            await _ticketTypeService.AddTicketTypeAsync(model);
+
+            return Ok();
+        }
+
+        [Authorize(Roles = "admin")]
+        [HttpPut]
+        public async Task<ActionResult<Response<string>>> Put([FromBody] TicketTypeViewModel model)
```

Comment:
If you're trying to follow best practices for RESTful API shouldn't put request have a bit different url? Is there something missing here?